scripts/Muon/GUI/ElementalAnalysis2/load_widget/load_models.py

```diff
@@ -0,0 +1,122 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantid.simpleapi import LoadElementalAnalysisData
+from Muon.GUI.Common.muon_load_data import MuonLoadData
+from Muon.GUI.ElementalAnalysis2.context.context import RunObject
+
+
+class LoadWidgetModel(object):
+    def __init__(self, loaded_data_store=MuonLoadData(), context=None):
+        self._loaded_data_store = loaded_data_store
+        self._data_context = context.data_context
+        self._context = context
+        return
+
+    @property
+    def runs(self):
+        return self._data_context.current_runs
+
+    @property
+    def current_runs(self):
+        return self._data_context.current_runs
+
+    @current_runs.setter
+    def current_runs(self, value):
+        self._data_context.current_runs = value
+
+    def clear_data(self):
+        self._loaded_data_store.clear()
+        self._data_context.current_runs = []
+
+
+class BrowseFileWidgetModel(object):
+    def __init__(self, loaded_data_store=MuonLoadData(), context=None):
+        self._loaded_data_store = loaded_data_store
+        self._data_context = context.data_context
+
+    @property
+    def loaded_runs(self):
+        return self._loaded_data_store.get_parameter("run")
+
+    @property
+    def current_runs(self):
+        return self._data_context.current_runs
+
+    @current_runs.setter
+    def current_runs(self, value):
+        self._data_context.current_runs = value
+
+    def clear(self):
+        self._loaded_data_store.clear()
+
+
+class LoadRunWidgetModel(object):
+    def __init__(self, loaded_data_store=MuonLoadData(), context=None):
+        self._loaded_data_store = loaded_data_store
+        self._data_context = context.data_context
+        self._runs = []
+        self._current_run = None
+        self._directory = ""
+
+    # Used with load thread
+    def loadData(self, runs):
+        self._runs = runs
+
+    # Used with load thread
+    def execute(self):
+        failed_files = []
+        for run in self._runs:
+            try:
+                groupws, self._directory = LoadElementalAnalysisData(Run=run, GroupWorkspace=str(run))
+                detectors = []
+                workspace_list = groupws.getNames()
+                for item in workspace_list:
+                    detector_name = item.split(';', 1)[-1].lstrip()
```

Comment:
lets add a comment explaining this line. It might be worth giving an example of the what we are starting with